### PR TITLE
updated util functions

### PR DESCRIPTION
**Pull request by rohitnarula7176**
*Updated a couple of util functions.

### CODE REVIEW

File: IdentityCore/src/util/NSString+MSIDExtensions.m [hunk 1]
```diff
@@ -99,33 +99,22 @@ - (NSString *)msidTrimmedString
...
-    
-    NSString *encodedString = [self stringByAddingPercentEncodingWithAllowedCharacters:set];
+    /*https://tools.ietf.org/html/rfc3986#section-3
+     Characters that are allowed in a URI but do not have a reserved
+     purpose are called unreserved.  These include uppercase and lowercase
+     letters, decimal digits, hyphen, period, underscore, and tilde.
+     unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
+     */
+    NSMutableCharacterSet *allowedCharacters = [NSMutableCharacterSet alphanumericCharacterSet];
+    NSString *unreserved = @"-._~ ";
```
Comment (oldalton) [inline IdentityCore/src/util/NSString+MSIDExtensions.m]: note: this now makes underscore a valid character while it wasn't before.
Also, creating character set only once is a nice optimization, not sure why it got removed.

Comment (oldalton) [reply IdentityCore/src/util/NSString+MSIDExtensions.m]: But you're doing the same - taking alphanumericCharacterSet and adding characters to it. One less operation then before, but doing it only once won't hurt. Regarding underscore, we need to print URLQueryAllowedCharacterSet and see if it's there or not.

Comment (rohitnarula7176) [reply IdentityCore/src/util/NSString+MSIDExtensions.m]: I have printed the URLQueryAllowedCharacterSet and confirmed that it contains '_'

File: IdentityCore/src/util/NSData+MSIDExtensions.m [hunk 2]
```diff
@@ -72,38 +72,19 @@ - (NSDictionary *)msidToJsonDictionary:(NSError **)error
...
-    
-    // Input string is not sized correctly to be base64 URL encoded.
-    
-    NSUInteger stringMod4 = base64encoded.length % 4;
-    
-    if (stringMod4 == 1)
-    {
-        return nil;
+    // Converts base64 no padding to base64 with padding
+    while (encodedStringCopy.length % 4 != 0) {
```
[the remark this answers is not in the document]
Comment (oldalton) [reply IdentityCore/src/util/NSData+MSIDExtensions.m]: I'll let @unpluggedk confirm this as he added the original code :)

**Pull request closed without merging**